modules/transport-netty4/src/test/java/org/elasticsearch/rest/discovery/Zen2RestApiIT.java

```diff
@@ -0,0 +1,152 @@
+                .put(IndexMetaData.SETTING_NUMBER_OF_SHARDS, nodeCount) // causes rebalancing
+                .put(IndexMetaData.SETTING_NUMBER_OF_REPLICAS, 1)
+                .build());
+        ensureGreen("test");
+
+        RestClient restClient = getRestClient();
+
+        internalCluster().rollingRestart(new InternalTestCluster.RestartCallback() {
+            @Override
+            public void doAfterNodes(int n, Client client) throws IOException {
+                ensureGreen("test");
+                Response response =
+                    restClient.performRequest(new Request("POST", "/_cluster/withdrawn_votes/" + internalCluster().getNodeNames()[n]));
+                assertThat(response.getStatusLine().getStatusCode(), is(200));
+            }
+
+            @Override
+            public Settings onNodeStopped(String nodeName) throws IOException {
+                String viaNode = randomValueOtherThan(nodeName, () -> randomFrom(nodes));
+
+                List<Node> allNodes = restClient.getNodes();
+                try {
+                    restClient.setNodes(
+                        Collections.singletonList(
+                            new Node(
+                                HttpHost.create(
+                                    internalCluster().getInstance(HttpServerTransport.class, viaNode)
+                                        .boundAddress().publishAddress().toString()
+                                )
+                            )
+                        )
+                    );
+                    Response deleteResponse = restClient.performRequest(new Request("DELETE", "/_cluster/withdrawn_votes"));
+                    assertThat(deleteResponse.getStatusLine().getStatusCode(), is(200));
+
+                    final ClusterHealthRequestBuilder clusterHealthRequestBuilder = client(viaNode).admin().cluster().prepareHealth()
+                        .setWaitForEvents(Priority.LANGUID)
+                        .setWaitForNodes(Integer.toString(nodeCount - 1))
+                        .setTimeout(TimeValue.timeValueSeconds(30L));
+
+                    clusterHealthRequestBuilder.setWaitForYellowStatus();
+                    ClusterHealthResponse clusterHealthResponse = clusterHealthRequestBuilder.get();
+                    assertFalse(nodeName, clusterHealthResponse.isTimedOut());
+                    return Settings.EMPTY;
+                } finally {
+                    restClient.setNodes(allNodes);
+                }
+            }
+        });
+        ensureStableCluster(nodeCount);
+        ensureGreen("test");
+        assertThat(internalCluster().size(), is(2));
+    }
+
+    public void testBasicRestApi() throws Exception {
+        List<String> nodes = internalCluster().startNodes(3);
+        RestClient restClient = getRestClient();
+        Response deleteResponse = restClient.performRequest(new Request("DELETE", "/_cluster/withdrawn_votes"));
```

Comment:
Think it'd make more sense to put this after the `POST`. There will, at some point, be an assertion that there's no voting tombstones in the cluster at the end of the test.

Reply:
I agree. Just tried making that change but running into this error as a  result:

```
[2018-11-29T00:04:21,996][WARN ][r.suppressed             ] [node_t0] path: /_cluster/withdrawn_votes, params: {}
org.elasticsearch.transport.RemoteTransportException: [node_t1][127.0.0.1:33009][cluster:admin/voting/clear_tombstones]
Caused by: org.elasticsearch.ElasticsearchTimeoutException: timed out waiting for removal of nodes; if nodes should not be removed, set waitForRemoval to false. [{node_t2}{ps-Qi3TfSAOMEW6D-A5uAA}]
	at org.elasticsearch.action.admin.cluster.configuration.TransportClearVotingTombstonesAction$1.onTimeout(TransportClearVotingTombstonesAction.java:109) ~[main/:?]
	at org.elasticsearch.cluster.ClusterStateObserver$ContextPreservingListener.onTimeout(ClusterStateObserver.java:322) ~[main/:?]
	at org.elasticsearch.cluster.ClusterStateObserver$ObserverClusterStateListener.onTimeout(ClusterStateObserver.java:249) ~[main/:?]
	at org.elasticsearch.cluster.service.ClusterApplierService$NotifyTimeout.run(ClusterApplierService.java:561) ~[main/:?]
	at org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingRunnable.run(ThreadContext.java:627) ~[main/:?]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128) ~[?:?]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628) ~[?:?]
	at java.lang.Thread.run(Thread.java:834) [?:?]
```

(no matter what node I add a Tombstone for, this happens)

> if nodes should not be removed, waitForRemoval to false

Should I do that in the Rest API?